middle_end/flambda/un_anf.ml

```diff
@@ -161,38 +161,38 @@ let make_var_info (clam : Clambda.ulambda) : var_info =
           captured_variables
       in
       let environment_vars =
-      List.fold_left (fun (
+      List.fold_left (fun environment_vars (
         { Clambda. label; arity; params; return; body; dbg; env; } as clos) ->
-          let environment_vars =
             (match closure_environment_var clos with
             | None -> environment_vars
             | Some env_var ->
               V.Set.add (VP.var env_var) environment_vars )
-          in
-          ignore_function_label label;
+          ignore_function_label label; 
           ignore_int arity;
           ignore_params_with_value_kind params;
           ignore_value_kind return;
           loop ~depth:(depth + 1) ~environment_vars:environment_vars body;
           ignore_debuginfo dbg;
           ignore_var_option env;
-          environment_vars)
-        in
-        functions;
-        environment_vars
+          )
+          environment_vars
+          clos
```

Comment:
`clos` is the name used to refer to each individual element (or closure). Here the argument you need to pass is the list of all such elements, which happens to be `functions`. You can then remove the `functions;` line below, which is not useful anymore.